test/e2e/vault-decryption-chrome.spec.ts

```diff
@@ -0,0 +1,298 @@
+
+  // Use the first of the `.log` files found
+  return path.resolve(extensionStoragePath, logFiles[0]);
+}
+
+/**
+ * Gets the size of a file in bytes.
+ *
+ * @param filePath - The path to the file.
+ * @returns A promise that resolves to the size of the file in bytes.
+ */
+async function getFileSize(filePath: string): Promise<number> {
+  const stats = await fs.promises.stat(filePath);
+  console.log(`File Size =========================: ${stats.size} bytes`);
+  return stats.size;
+}
+
+/**
+ * Retry logic to ensure Chrome has finish writing into the file.
+ *
+ * @param options - The options object.
+ * @param options.driver - The WebDriver instance.
+ * @param options.filePath - The path to the file.
+ * @param options.maxRetries - The maximum number of retries.
+ * @param options.minFileSize - The minimum file size in bytes.
+ * @returns
+ */
+async function waitUntilFileIsWritten({
+  driver,
+  filePath,
+  maxRetries = 3,
+  minFileSize = 1000000,
+}: {
+  driver: Driver;
+  filePath: string;
+  maxRetries?: number;
+  minFileSize?: number;
+}): Promise<void> {
+  for (let attempt = 0; attempt < maxRetries; attempt++) {
+    const fileSize = await getFileSize(filePath);
+    if (fileSize > minFileSize) {
+      break;
+    } else {
+      console.log(`File size is too small (${fileSize} bytes)`);
+      if (attempt < maxRetries - 1) {
+        console.log(`Waiting for 2 seconds before retrying...`);
+        await driver.delay(2000);
+      }
+    }
+  }
+}
+
+/**
+ * Closes the announcements popover if present
+ *
+ * @param driver
+ */
+async function closePopoverIfPresent(driver: Driver) {
+  const popoverButtonSelector = '[data-testid="popover-close"]';
+  // It shows in the Smart Transactions Opt-In Modal.
+  const enableButtonSelector = {
+    text: 'Enable',
+    tag: 'button',
+  };
+  await driver.clickElementSafe(popoverButtonSelector);
+  await driver.clickElementSafe(enableButtonSelector);
+
+  // Token Autodetection Independent Announcement
+  const tokenAutodetection = {
+    css: '[data-testid="auto-detect-token-modal"] button',
+    text: 'Not right now',
+  };
+  await driver.clickElementSafe(tokenAutodetection);
+
+  // NFT Autodetection Independent Announcement
+  const nftAutodetection = {
+    css: '[data-testid="auto-detect-nft-modal"] button',
+    text: 'Not right now',
+  };
+  await driver.clickElementSafe(nftAutodetection);
+}
+
+describe('Vault Decryptor Page', function () {
+  it('is able to decrypt the vault uploading the log file in the vault-decryptor webapp', async function () {
+    await withFixtures(
+      {
+        disableServerMochaToBackground: true,
+      },
+      async ({ driver }) => {
+        // we don't need to use navigate since MM will automatically open a new window in prod build
+        await driver.waitUntilXWindowHandles(2);
+
+        // we cannot use the customized driver functionsas there is no socket for window communications in prod builds
```

Comment:
there is minor typo now we have approvals we could correct later. Able to understand too.

Reply:
Looks like the PR is merged..

Reply:
Sure thanks @chloeYue

Reply:
addressed here https://github.com/MetaMask/metamask-extension/pull/28424/commits/902a275700812cef92a93faaa6e8f16eac93e089